src/parsing.jl

```diff
@@ -387,6 +387,62 @@ fun_to_vec(ex::QuoteNode;
            gensym_names::Bool=false,
            outer_flags::Union{NamedTuple, Nothing}=nothing) = ex
 
+
+"""
+    rename_kw_to_pair(ex::Expr)
+
+Given an expression where the left- and right- hand side 
+both are both valid column identifiers,  i.e., a `QuoteNode`
+or an expression beginning with `$DOLLAR`, or a "full" expression of the form 
+`$DOLLAR(:x => :y)`, return an expression, where expression arguments of type 
+`QuoteNode`` are converted to `String``.
+"""           
+function rename_kw_to_pair(ex::Expr)
+    
+    ex_col = get_column_expr(ex)        
+    if ex_col !== nothing                 
+        return ex_col
+    end
+    
+    lhs = let t = ex.args[1]
+        
+        s = get_column_expr(t)
+        if s === nothing
+            throw(ArgumentError("Invalid column identifier on LHS in DataFramesMeta.jl macro"))
+        end
+        
+        s
+    end
+    
+    rhs = ex.args[2]    
+    rhs_col = get_column_expr(rhs)
+    
+    if rhs_col === nothing
+        throw(ArgumentError("Invalid column identifier on RHS in DataFramesMeta.jl macro"))
+    end
+
+    if rhs_col !== nothing
+        src = rhs_col
+        dest = lhs        
+        return :($src => $dest)
+    end
+
+end
+
+function pairs_to_str_pairs(args...)
+    
+    map(args) do arg        
+        if !(arg isa Pair)            
+            throw(ArgumentError("Non-pair created in @rename"))
+        end         
+
+        if first(arg) isa Int            
+            return first(arg) => string(last(arg))
```

Comment:
@MatthewRGonzalez 

Thank you for your work. One small thing. I think `string` is too strong here. Here's the issue. `:(AsTable(:x))` passes `get_column_expr` because we allow `AsTable` to be used as a column identifier identifier. inside transformations. So you end up calling `string(:old_col) => string(AsTable(:x))`

The problem is that `string(AsTable(:x))` gets turned into `"AsTable(x)"`. 

The solution is two-fold

1. Make a new function called `get_column_expr_rename` which only looks for `:x` and `$(...)`, no `AsTable`
2. Add more error handling in the `pairs_to_str_pairs` function. 

I've done both and added more tests. I've also re-written the docstring. 

So I think this is good to merge!

Reply:
Good catch, @pdeffebach. Thanks for adding the style fixes, error handling, and adding `get_column_expr_rename` as well!